assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -1,221 +1,161 @@
+			: Math.max( selectedImageIdIndex - 1, 0 );
+	context.selectedImage = imagesIds[ nextImageIndex ];
+};
+
+const productGallery = {
+	state: {
+		get isSelected() {
+			const { selectedImage, imageId } = getContext();
+			return selectedImage === imageId;
+		},
+		get pagerDotFillOpacity(): number {
+			return state.isSelected ? 1 : 0.2;
+		},
+	},
+	actions: {
+		closeDialog: () => {
+			const context = getContext();
+			context.isDialogOpen = false;
+		},
+		openDialog: () => {
+			const context = getContext();
+			context.isDialogOpen = true;
+		},
+		selectImage: () => {
+			const context = getContext();
+			context.selectedImage = context.imageId;
+		},
+		selectNextImage: ( event: MouseEvent ) => {
+			event.stopPropagation();
+			const context = getContext();
+			selectImage( context, 'next' );
+		},
+		selectPreviousImage: ( event: MouseEvent ) => {
+			event.stopPropagation();
+			const context = getContext();
+			selectImage( context, 'previous' );
+		},
+	},
+	callbacks: {
+		watchForChangesOnAddToCartForm: () => {
+			const context = getContext();
+			const variableProductCartForm = document.querySelector(
+				`form[data-product_id="${ context.productId }"]`
+			);
+
+			if ( ! variableProductCartForm ) {
+				return;
+			}
+
+			// TODO: Replace with an interactive block that calls `actions.selectImage`.
+			const observer = new MutationObserver( function ( mutations ) {
+				for ( const mutation of mutations ) {
+					const mutationTarget = mutation.target as HTMLElement;
+					const currentImageAttribute =
+						mutationTarget.getAttribute( 'current-image' );
+					if (
+						mutation.type === 'attributes' &&
+						currentImageAttribute &&
+						context.visibleImagesIds.includes(
+							currentImageAttribute
+						)
+					) {
+						context.selectedImage = currentImageAttribute;
+					}
 				}
+			} );
 
-				const observer = new MutationObserver( function ( mutations ) {
-					for ( const mutation of mutations ) {
-						const mutationTarget = mutation.target as HTMLElement;
-						const currentImageAttribute =
-							mutationTarget.getAttribute( 'current-image' );
-						if (
-							mutation.type === 'attributes' &&
-							currentImageAttribute &&
-							store.context.woocommerce.visibleImagesIds.includes(
-								currentImageAttribute
-							)
-						) {
-							store.context.woocommerce.selectedImage =
-								currentImageAttribute;
-						}
-					}
-				} );
+			observer.observe( variableProductCartForm, {
+				attributes: true,
+			} );
 
-				observer.observe( variableProductCartForm, {
-					attributes: true,
-				} );
+			return () => {
+				observer.disconnect();
+			};
+		},
+		keyboardAccess: () => {
+			const context = getContext();
+			let allowNavigation = true;
 
-				return () => {
-					observer.disconnect();
-				};
-			},
-			keyboardAccess: ( store: Store ) => {
-				const { context, actions } = store;
-				let allowNavigation = true;
+			const handleKeyEvents = ( event: Event ) => {
```

Comment:
Instead of the custom interface, it would be better to use `KeyboardEvent` here, right? 🤔